### PR TITLE
MES 4182: Fixed deprecated usage of buffer

### PR DESCRIPTION
**Pull request by faggc**
+ Fixed deprecated usage of isUndefined
+ Added linter rule to locate deprecated code

# Description and relevant Jira numbers

## Pull Request checklist

- [ ] [WIP] tag removed from PR title
- [ ] PR has an understandable description

## Git feature branch checklist

- [ ] branch name comply with our branching strategy
- [ ] git branch contains relevant JIRA ticket number
- [ ] branch rebased against the latest develop

## Sign off process checklist

- [ ] Code has been tested manually
- [ ] Tested by QA
- [ ] PO's approval
- [ ] Reviewers selected in Github
- [ ] PR link added to JIRA

### CODE REVIEW

File: src/functions/uploadToRSIS/framework/repo/rsis-repository.ts [hunk 1]
```diff
@@ -27,7 +26,7 @@ export const saveTestResult = async (
   const params = mappedFields.map(field => field.val);
 
   if (config.useRSIS) {
-    if (isUndefined(connection)) {
+    if (connection === undefined) {
```
Comment (rossfauldsbjss) [inline src/functions/uploadToRSIS/framework/repo/rsis-repository.ts:29]: why the removal of the util function?

**Review by rossfauldsbjss — APPROVED** [review]
Wondering about the removal of the isUndefined method but looks good.

**Review by TomBillingtonUK — APPROVED** [review]

**Comment by faggc** [comment]
@rossfauldsbjss isUndefined was identified as deprecated by the linter and this was the suggested solution

File: src/functions/uploadToRSIS/framework/repo/rsis-repository.ts (hunk 1 above)
Comment (faggc) [reply src/functions/uploadToRSIS/framework/repo/rsis-repository.ts:29]: https://github.com/nodejs/node-v0.x-archive/issues/8235
The function has been deprecated

**Pull request merged**